### ISSUE TITLE
typeof type guards don't work with switch statements

### ISSUE THREAD

**Issue by Gozala** [Microsoft/TypeScript#16068]
**TypeScript Version:**  2.3

**Code**

```ts
declare function parse(input: string): boolean | string | number

const data = parse('whatever')

switch (typeof data) {
    case 'number':
        console.log(data.toFixed())
    case 'string':
        console.log(data.toUpperCase())
    case 'boolean':
        console.log(!data)
}
```

**Expected behavior:**

I would expect `data` type to be refined in each case & for this code to check.

**Actual behavior:**

`data` type does not seems to get refined by switch cases, there for errors are reported when data is treated as number and as string:

```
Property 'toFixed' does not exist on type 'string | number | boolean'.
  Property 'toFixed' does not exist on type 'string'.

Property 'toUpperCase' does not exist on type 'string | number | boolean'.
  Property 'toUpperCase' does not exist on type 'number'.
```

**Comment by RyanCavanaugh** [comment]
Duplicate #2214.
![image](https://cloud.githubusercontent.com/assets/6685088/26428339/fc090432-4095-11e7-9318-2018e309c24d.png)
